src/test/java/com/ot/restaurant/entities/IoCContainerTest.java

```diff
@@ -0,0 +1,85 @@
+package com.ot.restaurant.entities;
+
+import org.junit.jupiter.api.Test;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class IoCContainerTest {
+
+    @Test
+    void singletonIoCContainer() {
+
+        IoCContainer ioc = IoCContainer.getInstance();
+        IoCContainer ioc2 = IoCContainer.getInstance();
+
+        assertEquals(ioc, ioc2);
+    }
+
+    @Test
+    void testRegisterAndResolve() throws Exception {
+        IoCContainer ioc = IoCContainer.getInstance();
+
+        String key = "keyEntiti";
+        Object value = new Object();
+
+        ioc.register(key, value);
+
+        Object valueResolvedIoC = ioc.resolve(key);
+
+        assertEquals(value, valueResolvedIoC);
+    }
+
+    @Test
+    void testDuplicated() throws Exception {
```

Comment:
Change the name according to above comment

Reply:
done